test/arbitrage.test.js

```diff
@@ -70,18 +69,28 @@ test('tapply vector char index', t => {
   t.true(r.all(r.is_equal(act,exp)))
 })
 
+test('tapply dataframe int index', t => {
+  var df = r.dataframe([1,1,2,2,2], [1,2,3,4,5], {"colnames": ["idx", "num"]})
+  var act = r.tapply(df.num, df.idx, r.sum)
+  var exp = [ 3, 12 ]
+  t.true(r.all(r.is_equal(act,exp)))
+})
 
-test.todo('tapply dataframe int index')
-test.todo('tapply dataframe char index')
+test('tapply dataframe char index', t => {
+  var df = r.dataframe(['a','a','b','b','b'], [1,2,3,4,5], {"colnames": ["idx", "num"]})
+  var act = r.tapply(df.num, df.idx, r.sum)
+  var exp = [ 3, 12 ]
+  t.true(r.all(r.is_equal(act,exp)))
+})
 
+test.todo('tapply throws an error when given a dataframe index')
```

Comment:
I like the names of these tests. The names of the tests above should be as descriptive.

Reply:
Done!